stumpy/core.py

```diff
@@ -471,19 +469,27 @@ def mass(Q, T, M_T=None, Σ_T=None):
 
     Q = np.asarray(Q)
     check_dtype(Q)
-    check_nan(Q)
 
     if Q.ndim != 1:  # pragma: no cover
         raise ValueError(f"Q is {Q.ndim}-dimensional and must be 1-dimensional. ")
+    m = Q.shape[0]
+
     T = np.asarray(T)
     check_dtype(T)
     check_nan(T)
 
     if T.ndim != 1:  # pragma: no cover
         raise ValueError(f"T is {T.ndim}-dimensional and must be 1-dimensional. ")
+    n = T.shape[0]
+
+    if np.any(np.isnan(Q)):
+        distance_profile = np.empty(n - m + 1)
+        distance_profile[
```

Comment:
We should squash these three lines into a single line and remove the comment. I think this is pretty self-explanatory and is consistent with the rest of the code base